### PR TITLE
Migration Shell commands for default value and null

### PR DESCRIPTION
**Pull request by cercos**
In response to this [issue](https://github.com/cakephp/migrations/issues/203)
Adds ability to set a field default value and whether it is nullable from the shell using the bake migrations command.

New syntax - fieldName[default_value:null]:fieldType[length]:indexType:indexName

### CODE REVIEW

**Comment by HavokInspiration** [comment]
Hi @cercos and thanks for your contribution.
I will review your pull request later on and will get back to you on this.

Looking at [the travis build](https://travis-ci.org/cakephp/migrations/builds/127122404), your changes seem to have broken the tests. They should be passing before this can be merged. We can give you a hand if you need to !

**Comment by HavokInspiration** [comment]
We'll also need some new tests to cover the feature. We can give a hand on this too.

**Comment by cercos** [comment]
I could use a hand on what needs to be fixed..this is the first time submitting a pull request.

**Comment by HavokInspiration** [comment]
First, you need to check out the Travis build which runs tests on various PHP versions and database drivers : https://travis-ci.org/cakephp/migrations/builds/127122404

Picking [the first one](https://travis-ci.org/cakephp/migrations/jobs/127122405) in the list, you can see that the changes you made break the tests because somehow, the length is put in the ``default`` key.

You will also have to add tests to [this test method](https://github.com/cakephp/migrations/blob/master/tests/TestCase/Shell/Task/MigrationTaskTest.php#L65-L110) to assert that the new syntax works as expected.

On a side note, @lorenzo, @josegonzalez : what do you guys think of the syntax ?

File: src/Util/ColumnParser.php [hunk 1]
```diff
@@ -36,11 +44,15 @@ public function parseFields($arguments)
...
         foreach ($arguments as $field) {
+            $fieldOpts = $this->getDefaults($field);
+
             preg_match($this->regexpParseColumn, $field, $matches);
             $field = $matches[1];
+
             $type = Hash::get($matches, 2);
             $indexType = Hash::get($matches, 3);
 
+
```
Comment (HavokInspiration) [inline src/Util/ColumnParser.php]: Extra new-line is not needed

File: src/Util/ColumnParser.php [hunk 2]
```diff
@@ -178,6 +191,30 @@ public function getTypeAndLength($field, $type)
...
     /**
+     * Get the defaults of a field based on the field passed
+     *
+     * @param string $field Name of field
+     * @return array First value is the default value for the field, second value is the if field is null. If no length
+     * can be extracted, null is returned for default value and false for null
+     */
+    public function getDefaults($field)
+    {
+
```
Comment (HavokInspiration) [inline src/Util/ColumnParser.php]: Extra new line is not needed

File: src/Util/ColumnParser.php [hunk 3]
```diff
@@ -178,6 +191,30 @@ public function getTypeAndLength($field, $type)
...
+        if (preg_match($this->regexpParseFieldDefaults, $field, $matches)) {
+            $options = explode(':', trim($matches[0], '[]'));
+            if (isset($options[1])) {
+                if ($options[1] == 'null') $matches['null'] = true;
```
Comment (HavokInspiration) [inline src/Util/ColumnParser.php]: Inline ``if`` is not allowed as per [our coding standards](http://book.cakephp.org/3.0/en/contributing/cakephp-coding-conventions.html)

File: src/Util/ColumnParser.php [hunk 4]
```diff
@@ -178,6 +191,30 @@ public function getTypeAndLength($field, $type)
...
+            }
+            $matches['default'] = ($options[0] == 'null' ? null : $options[0]);
+            unset($matches[0]);
+        }
+        if(!isset($matches['default'])) $matches['default'] = null;
```
Comment (HavokInspiration) [inline src/Util/ColumnParser.php]: Same comment as above

File: src/Util/ColumnParser.php [hunk 5]
```diff
@@ -178,6 +191,30 @@ public function getTypeAndLength($field, $type)
...
+        if(!isset($matches['null'])) $matches['null'] = false;
```
Comment (HavokInspiration) [inline src/Util/ColumnParser.php]: Same comment as above

File: src/Util/ColumnParser.php [hunk 6]
```diff
@@ -178,6 +191,30 @@ public function getTypeAndLength($field, $type)
...
+        return $matches;
+    }
+
+
```
Comment (HavokInspiration) [inline src/Util/ColumnParser.php]: Extra new line

**Comment by HavokInspiration** [comment]
I also encourage you to read the [coding standards we use](http://book.cakephp.org/3.0/en/contributing/cakephp-coding-conventions.html). This plugin more generally follows PSR2.

Coding standards are also inforced using the build on travis. You can find the report for your last commits [here](https://travis-ci.org/cakephp/migrations/jobs/127122420)

**Comment by HavokInspiration** [comment]
Closing in favor for #249 for the nullable part.

**Pull request closed without merging**